src/dfx/src/util/mod.rs

```diff
@@ -224,6 +230,39 @@ pub fn blob_from_arguments(
                             .context("Failed to create idl args.")?;
                         eprintln!("Sending the following random argument:\n{}\n", args);
                         args.to_bytes_with_types(env, &func.args)
+                    } else if is_terminal {
+                        use candid_parser::assist::{input_args, Context};
+                        let mut ctx = Context::new(env.clone());
+                        if let Some(env) = dfx_env {
+                            let principals = gather_principals_from_env(env);
+                            if !principals.is_empty() {
+                                let mut map = BTreeMap::new();
+                                map.insert("principal".to_string(), principals);
+                                ctx.set_completion(map);
+                            }
+                        }
+                        if is_init_arg {
+                            eprintln!(
+                                "This canister requires the following initialization argument."
+                            );
+                        } else {
+                            eprintln!("This method requires the following arguments.");
+                        }
+                        let args = input_args(&ctx, &func.args)?;
+                        eprintln!("Sending the following argument:\n{}\n", args);
+                        if is_init_arg {
+                            eprintln!(
+                                "Do you want to initialize canister with this argument? [y/N]"
```

Comment:
either "initialize the canister" or "initialize canister {name}" for correct grammar